core/src/service/integrity_service.rs

```diff
@@ -24,7 +25,16 @@ pub enum TestRepoError {
     Core(CoreError),
 }
 
-pub fn test_repo_integrity(config: &Config) -> Result<(), TestRepoError> {
+#[derive(Debug, Clone)]
+pub enum Warning {
+    EmptyFile(Uuid),
+    InvalidUTF8(Uuid),
```

Comment:
One check missing here, could be my fault for not tracking it anywhere:

if we're dealing with a `.draw` attempt to call the corresponding read_drawing sort of function (either in `drawing_service` or `lib.rs` and if the drawing is invalid, report that as it's own warning.